frontend/src/components/worksheets/BundleDetail/BundleStateTable/index.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
+import { FINAL_BUNDLE_STATES } from '../../../../constants';
+import { BundleFieldTable, BundleFieldRow } from '../BundleFieldTable/';
+import BundleStateBox from './BundleStateBox';
+
+/**
+ * This component renders a dynamic BundleFieldTable that changes as a bundle's
+ * state changes.
+ *
+ * If the bundle is in a final state, a simple state box will be shown.
+ * If the bundle is not in a final state, a state diagram will be shown.
+ */
+class BundleStateTable extends React.Component {
+    constructor(props) {
+        super(props);
+    }
+
+    roundTime(time) {
+        return Math.abs(Math.round(time));
+    }
+
+    getTimePassed() {
```

Comment:
Can we render the time field (which is time spent actually running)?  Time since created is also meaningful, but sometimes not the thing you want.  Maybe there's a way to render both?  In general, maybe the thing to do is to render a total time and the time spent in the current stage?

Also, if we have a function for rendering a duration, could we factor this out?  Ideally, it would match the one in `formatting.py`

Reply:
I have removed the time-since-creation logic altogether. I feel like having that plus running time may be confusing to the user.

I have added times under the `running` state and `preparing` state. Appending times to the other states seemed like the logic required wouldn't be worth the payoff but plz lmk your thoughts.

**Note:** Times are also rendered in the `More Details` section for users to reference later if needed.

Reply:
Logically this is great.  Would it be possible to move the times to the (right) side of the states, so that the state flowchart doesn't jump around when times are added?

Reply:
@percyliang certainly. Updated 👍 

https://user-images.githubusercontent.com/25855750/180668812-e2fcf78d-7f90-4f75-995d-3ccf5688374c.mov